### DISCUSSION TITLE
[Help wanted]Any docs to describe how to limit the syscall or file/dir the wasm code can use?

### DISCUSSION THREAD

**Discussion by pkking** [WasmEdge/WasmEdge#1140] [Q&A] [answered]
First of all, wasmedge is a cool project and happy spring festival.
After reading the (docs)[https://wasmedge.org/book/en], i do want to find if there's a way to controll which syscall or file can be used by the wasm code embed in my c program, but it seems there's not a page about this topic.

Did i miss something, or any suggestions?

**Comment by hydai** [comment, marked as answer]
Hi @pkking 
The WasmEdge tool (wasmedge and wasmedgec) enables WASI support by default.

If you don't want the WASI syscall, then you can use our WasmEdge C API and remove the WASI in the configuration. Reference: https://wasmedge.org/book/en/embed/c/ref.html#configurations

And you can also `WasmEdge_ImportObjectInitWASI` to limit the mapping folders and files. Reference: https://github.com/WasmEdge/WasmEdge/blob/master/include/api/wasmedge/wasmedge.h#L2405-L2408

**Reply by pkking** [reply]
thanks for your reply
